cluster-controller/src/test/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperationsMockIT.java

```diff
@@ -0,0 +1,284 @@
+            .endMetadata()
+            .withData(map(KafkaCluster.KEY_REPLICAS, "7",
+                    KafkaCluster.KEY_STORAGE, "{\"type\": \"ephemeral\"}",
+                    ZookeeperCluster.KEY_REPLICAS, "3",
+                    ZookeeperCluster.KEY_STORAGE, "{\"type\": \"ephemeral\"}"))
+            .build();
+
+    @Before
+    public void before() {
+        this.vertx = Vertx.vertx();
+    }
+
+    @After
+    public void after() {
+        this.vertx.close();
+    }
+
+    private static <T> Map<T, T> map(T... pairs) {
+        if (pairs.length % 2 != 0) {
+            throw new IllegalArgumentException();
+        }
+        Map<T, T> result = new HashMap<>(pairs.length / 2);
+        for (int i = 0; i < pairs.length; i += 2) {
+            result.put(pairs[i], pairs[i + 1]);
+        }
+        return result;
+    }
+
+    // TODO Test with absent storage key
+    // TODO Test with storage key lacking a type
+
+    private KafkaClusterOperations createCluster(TestContext context, KubernetesClient mockClient) {
+        ConfigMapOperations cmops = new ConfigMapOperations(vertx, mockClient);
+        ServiceOperations svcops = new ServiceOperations(vertx, mockClient);
+        KafkaSetOperations ksops = new KafkaSetOperations(vertx, mockClient);
+        ZookeeperSetOperations zksops = new ZookeeperSetOperations(vertx, mockClient);
+        DeploymentOperations depops = new DeploymentOperations(vertx, mockClient);
+        PvcOperations pvcops = new PvcOperations(vertx, mockClient);
+        KafkaClusterOperations kco = new KafkaClusterOperations(vertx, true, 2_000,
+                cmops, svcops, zksops, ksops, pvcops, depops);
+
+        LOGGER.info("Reconciling initially -> create");
+        Async createAsync = context.async();
+        kco.createOrUpdate(NAMESPACE, cluster.getMetadata().getName(), ar -> {
+            if (ar.failed()) ar.cause().printStackTrace();
+            context.assertTrue(ar.succeeded());
+            createAsync.complete();
+        });
+        createAsync.await();
+        return kco;
+    }
+
+    /** Create a cluster from a Kafka Cluster CM */
+    @Test
+    public void testCreateUpdateDelete(TestContext context) {
+        String clusterName = cluster.getMetadata().getName();
+        KubernetesClient mockClient = new MockKube().withInitialCms(Collections.singleton(cluster)
+        ).build();
+
+        KafkaClusterOperations kco = createCluster(context, mockClient);
+        LOGGER.info("Reconciling again -> update");
+        Async updateAsync = context.async();
+        kco.createOrUpdate(NAMESPACE, clusterName, ar -> {
+            if (ar.failed()) ar.cause().printStackTrace();
+            context.assertTrue(ar.succeeded());
+            updateAsync.complete();
+        });
+        updateAsync.await();
+        LOGGER.info("Reconciling again -> delete");
+        mockClient.configMaps().inNamespace(NAMESPACE).withName(clusterName).delete();
```

Comment:
What's the reason for using method `inNamespace()` in this case because you created mock `KubernetesClient mockClient` by `ConfigMap cluster` that already has NAMESPACE?